lib/src/main/java/com/basistheory/android/service/HttpClient.kt

```diff
@@ -0,0 +1,87 @@
+import com.basistheory.android.service.HttpMethod
+import com.basistheory.android.util.*
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.*
+import java.io.IOException
+
+interface Client {
+    suspend fun get(url: String, headers: Map<String, String>? = emptyMap()): Any?
+    suspend fun post(url: String, body: Any?, headers: Map<String, String>? = emptyMap()): Any?
+    suspend fun put(url: String, body: Any?, headers: Map<String, String>? = emptyMap()): Any?
+    suspend fun patch(url: String, body: Any?, headers: Map<String, String>? = emptyMap()): Any?
+    suspend fun delete(url: String, headers: Map<String, String>? = emptyMap()): Any?
+}
+
+class HttpClient(private val dispatcher: CoroutineDispatcher = Dispatchers.IO) : Client {
+    private val client: OkHttpClient = OkHttpClient()
+
+    override suspend fun post(url: String, body: Any?, headers: Map<String, String>?): Any =
+        withContext(dispatcher) {
+            send(url, HttpMethod.POST, body, headers)
+        }
+
+    override suspend fun get(url: String, headers: Map<String, String>?): Any? =
+        withContext(dispatcher) {
+            send(url, HttpMethod.GET, null, headers)
+        }
+
+    override suspend fun put(url: String, body: Any?, headers: Map<String, String>?): Any? =
+        withContext(dispatcher) {
+            send(url, HttpMethod.PUT, body, headers)
+        }
+
+    override suspend fun patch(url: String, body: Any?, headers: Map<String, String>?): Any? =
+        withContext(dispatcher) {
+            send(url, HttpMethod.PATCH, body, headers)
+        }
+
+    override suspend fun delete(url: String, headers: Map<String, String>?): Any? =
+        withContext(dispatcher) {
+            send(url, HttpMethod.DELETE, null, headers)
+        }
+
+    private fun requestBuilder(
```

Comment:
I would rename this to buildRequest since it is building it already

Reply:
Renamed ✅